cmd/juju/application/deployer/bundlehandler.go

```diff
@@ -380,6 +377,76 @@ func (h *bundleHandler) resolveCharmsAndEndpoints() error {
 	return nil
 }
 
+func (h *bundleHandler) resolveCharmChannelAndRevision(charmURL, charmSeries, charmChannel, arch string) (string, int, error) {
+	if h.isLocalCharm(charmURL) {
+		return charmChannel, -1, nil
```

Comment:
If this is a charm that I downloaded for offline deployments won't it have a revision file injected? Does it make sense to pull the revision (if present) from there?

Reply:
Let me check this, but I believe that we lost the original path here when we created the bundle data and so we can't go and inspect the revision file.